python/graphstorm/eval/eval_func.py

```diff
@@ -740,6 +765,149 @@ def compute_fscore(y_preds, y_targets, beta):
 
     return fscore
 
+def compute_precision_at_recall(y_preds, y_targets, beta=1., weights=None):
+    """ compute precision at recall at beta for classification tasks.
+        If there is no a recall score equal to beta, it returns the precision
+            at the largest recall less than beta.
+        If there are multiple precision scores when recall equal to beta
+            or the largest recall less than beta, it returns the maximum precision
+            among these precision scores.
+        If any errors occur, raise the error to callers and stop.
+
+        It only supports binary classification.
+
+        Parameters
+        ----------
+        y_preds : Target scores in 1D or 2D tensor.
+        y_targets: Array-like of shape (n_samples,) or (n_samples, n_classes) True labels or
+                   binary label indicators.
+        weights: List of weights with the same number of classes in labels.
+        beta: Beta value of precision for getting recall. Default is 1.0.
+        Returns
+        -------
+        float: The precision_at_recall score.
+    """
+    if beta > 1:
+        logging.warning(
+            "WARNING: beta should be between 0 and 1, but got %s. Using 1. instead.", str(beta))
+    if beta < 0:
+        logging.warning(
+            "WARNING: beta should be between 0 and 1, but got %s. Using 0. instead.", str(beta))
+
+    y_true = y_targets.cpu().numpy()
+    y_pred = y_preds.cpu().numpy()
+
+    # only support binary classification
+    nclass = len(np.unique(y_true))
```

Comment:
There is a corner case that the `y_true` could be either ALL 0s or 1s, which will fail the assertion below, but still can correctly compute the precision and recall by using the `precision_recall_curve()`.